src/php/lib/Grpc/BaseStub.php

```diff
@@ -69,6 +69,18 @@ public function __construct($hostname, $opts, $channel = null)
             return;
         }
 
+        $this->channel = static::getDefaultChannel($opts);
+    }
+
+    /**
+     * Creates and returns the default Channel
+     *
+     * @param array $opts Channel constructor options
+     *
+     * @return Channel The channel
+     */
+    public static function getDefaultChannel(array $opts)
```

Comment:
Could also make this `public static function getDefaultChannel(ChannelCredentials $credentials, array $opts = [])` as the credentials are required.

Reply:
I think it's okay to leave it inside the `$opts`. C extensions will read `$opts['credentials']`.